### PR TITLE
Search ingestion test (cont)

### PR DESCRIPTION
**Pull request by qnkhuat**
Follow up of https://github.com/metabase/metabase/pull/50388 adding tests for the rest of the models

Ref #50261

### CODE REVIEW

File: src/metabase/models/segment.clj [hunk 1]
```diff
@@ -192,7 +192,7 @@
 (search/define-spec "segment"
   {:model        :model/Segment
    :attrs        {:archived      true
-                  :collection-id :id
+                  :collection-id false
```
Comment (qnkhuat) [inline src/metabase/models/segment.clj:195]: this was wrong right?

Comment (crisptrutski) [reply src/metabase/models/segment.clj:195]: 🤦

File: test/metabase/search/postgres/index_test.clj [hunk 2]
```diff
@@ -285,18 +315,200 @@
...
+                  :model_created_at yesterday
+                  :model_updated_at yesterday})
+                (ingest-then-fetch! "table" table-name)))))))
+
+(deftest collection-ingestion-test
+  (search.tu/with-temp-index-table
+    (let [collection-name (mt/random-name)
+          yesterday       (t/- (t/offset-date-time) (t/days 1))]
+      (mt/with-temp [:model/Collection {coll-id :id} {:name collection-name
+                                                      ;; :authority_level = "official"
```
Comment (crisptrutski) [inline test/metabase/search/postgres/index_test.clj:380]: This ends up mapping to `:official_collection true`

File: test/metabase/search/postgres/index_test.clj [hunk 3]
```diff
@@ -285,18 +315,200 @@
...
+                (ingest-then-fetch! "dashboard" dashboard-name)))))))
+
+(deftest segment-ingestion-test
+  (search.tu/with-temp-index-table
+    (let [segment-name (mt/random-name)
+          yesterday    (t/- (t/offset-date-time) (t/days 1))]
+      (mt/with-temp [:model/Database   {db-id :id}      {}
+                     :model/Table      {table-id :id}   {:db_id db-id}
+                     :model/Segment    {segment-id :id} {:name       segment-name
+                                                         ;; :table_id = table-id
```
Comment (crisptrutski) [inline test/metabase/search/postgres/index_test.clj:476]: Not asserting for this

File: test/metabase/search/postgres/index_test.clj [hunk 4]
```diff
@@ -285,18 +315,200 @@
...
+                                                                  :pk_ref     (mt/$ids :products $id)
+                                                                  :value_ref  (mt/$ids :products $title)
+                                                                  :schedule   "0 0 0 * * *"
+                                                                  :state      "initial"
+                                                                  :creator_id (mt/user->id :rasta)}]
+        ;; :model/ModelIndexValue does not have id column so does not work nicely with with-temp
+        (t2/insert! :model/ModelIndexValue {:name       entity-name
+                                            :model_index_id model-index-id
+                                            ;; :model_id = model-id
+                                            :model_pk   42})
```
Comment (crisptrutski) [inline test/metabase/search/postgres/index_test.clj:507]: nit: map values alignment

**Review by crisptrutski — APPROVED** [review]
This is great! Not gonna block on any of the small stuff, want to merge so we can start refactoring.

**Pull request merged**